src/ioc/container.ts

```diff
@@ -1,122 +1,48 @@
-import {getType, MaybeToken, stringifyToken} from "./token";
-import {NOPLUGINS} from "./symbol";
-
-interface Item<T> {
-    factory?: Factory<T>;
-    value?: Value<T>;
-    cache?: T;
-    singleton?: boolean;
-    plugins: Plugin<T>[];
-}
-
-export type Plugin<Dependency = any, Target = any> = (
```

Comment:
I think we can change this to 
```
export type Plugin<Dependency = unknown> = (
    dependency: Dependency,
    target: unknown,
    tags: symbol[],
    token: MaybeToken<Dependency>,
    container: Container,
) => void;
```
We can not sure that the target is what we expect. I also could not set it in my test project:

```
export const serviceListenerPlugin: Plugin<Listenable, Component> = (service, component, args) => { 

// then this fails:
        .withPlugin(serviceListenerPlugin);
/*
TS2345: Argument of type 'Plugin<Listenable, Component<{}, {}>>' is not assignable to parameter of type 'Plugin<TranslatorInterface, unknown>'.
  Type 'unknown' is not assignable to type 'Component<{}, {}>'.
*/
```

Reply:
currently it is:
```
export type Plugin<Dependency = unknown, Target = unknown> = ( 
```

Reply:
Yes, unfortunately we can know what it will be. and the second type is never used inside the library 👍